scipy/optimize/_qap.py

```diff
@@ -0,0 +1,709 @@
+            adjacency matrices. In the literature, a partial match is also
+            known as a "seed" [2]_.
+
+            Each row of `partial_match` specifies the indices of a pair of
+            corresponding nodes, that is, node ``partial_match[i, 0]`` is
+            matched to node ``partial_match[i, 1]``. Accordingly,
+            ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+            less than the number of nodes.
+
+        maximize : bool (default = False)
+            Setting `maximize` to ``True`` solves the Graph Matching Problem
+            (GMP) rather than the Quadratic Assingnment Problem (QAP).
+
+        rng : {None, int, `~np.random.RandomState`, `~np.random.Generator`}
+            This parameter defines the object to use for drawing random
+            variates.
+            If `rng` is ``None`` the `~np.random.RandomState` singleton is
+            used.
+            If `rng` is an int, a new ``RandomState`` instance is used,
+            seeded with `rng`.
+            If `rng` is already a ``RandomState`` or ``Generator``
+            instance, then that object is used.
+            Default is None.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` containing the following
+        fields.
+
+        col_ind : 1-D array
+            An array of column indices corresponding with the best
+            permutation of the nodes of `B` found.
+        score : float
+            The corresponding value of the objective function.
+        nit : int
+            The number of Franke-Wolfe iterations performed during
+            the initialization resulting in the permutation
+            returned.
+
+    Notes
+    -----
+    The default method :ref:`'faq' <optimize.qap-faq>` uses the Fast
+    Approximate QAP algorithm [1]_; it is typically offers the best
+    combination of speed and accuracy.
+    Method :ref:`'2opt' <optimize.qap-2opt>` can be computationally expensive,
+    but may be a useful alternative, or it can be used to refine the solution
+    returned by another method.
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015,
+           https://doi.org/10.1371/journal.pone.0121002
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, "Seeded graph matching", Pattern Recognit. 87 (2019):
+           203-215, https://doi.org/10.1016/j.patcog.2018.09.014
+
+    .. [3] "2-opt," Wikipedia.
+           https://en.wikipedia.org/wiki/2-opt
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.optimize import quadratic_assignment
+    >>> cost = np.array([[0, 80, 150, 170], [80, 0, 130, 100],
```

Comment:
Can we call these (`cost` and `dist`) `A` and `B`, so that the relationship with the formula below is clearer?
I prefer `A` and `B` by the way, since they are not problem-specific.

Reply:
OK, pushing with that change soon.

Reply:
Done.